Add -function option and user specified arguments

arianneb opened the pull request:
Add option to take command line arguments for user specified function. I tried to remove the white space changes at the end of this commit for like 20 minutes. It's not happening. Sorry.

Commit:
Add -function option / parse user specified arguments

Parse command line arguments into usrArgs vector

Signed-off-by: Arianne Butler <arianne.butler@ibm.com>

rwy7 commented on b9run/main.cpp:
```diff
@@ -86,14 +96,9 @@ static bool parseArguments(RunConfig& cfg, const int argc, char* argv[]) {
-    if (i < argc) {
-      cfg.mainFunction = argv[i++];
-    }
+    cfg.moduleName = argv[i];
```
This should still be `argv[i++];`

rwy7 commented:
So far this is looking good. I think instead of `std::vector<int>`, you should use `std::vector<StackElement>` for usrArgs.

Don't forget to update the usage string as well 👍

Commit:
Pop CL args onto the stack

Check size of argument vector, throw BadFunctionCallException if
incorrect number of arguments are passed.
Refactor vm.run functions to accept argument vector as parameter.

Signed-off-by: Arianne Butler <arianne.butler@ibm.com>

xliang6 reviewed (dismissed):
Can you change the commit title since this is a public repository? Sorry to nitpick, thanks.

rwy7 commented on b9/core.cpp:
```diff
@@ -352,24 +352,35 @@ void ExecutionContext::reset() {
-    std::cout << "Pushing arg[" << i << "] = " << arg << std::endl;
-    executionContext_.push(arg);
+  if (usrArgs.size() == 0) {
+    // push random arguments to send to the program
```
don't push random args, just delete this branch,

arianneb replied:
@rwy0717 what if the user doesn't specify arguments for a function the requires them? error out?

rwy7 replied:
Yep, throw an exception, and catch it in b9run.
